tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from adhocracy4.comments.models import Comment
+from adhocracy4.comments_async.serializers import CommentSerializer
+
+
+@pytest.mark.django_db
+def test_serializer(client, comment_factory):
+
+    comment_factory.create(pk=1)
+    comment_factory.create(pk=2, is_blocked=True)
+    comment_factory.create(pk=3, is_censored=True)
+    comment_factory.create(pk=4, is_removed=True)
+    comment_factory.create(pk=5, is_moderator_marked=True)
+
+    comment_list = Comment.objects.all().order_by('pk')
+
+    comment_serializer = CommentSerializer(comment_list, many=True)
+
+    # comment_data = comment_serializer.data
```

Comment:
This line fails, as trace shows 
*** KeyError: 'request'
with a reference to [line 123 in the serializer](https://github.com/liqd/adhocracy4/blob/3a30aa9cd701416516f41a66e731151f70b51aea/adhocracy4/comments_async/serializers.py#L123):
        user = self.context['request'].user 
after the test has run.

@fuzzylogic2000 I can't reference the data for other assert statements.

Would I have to add someting like
`apiclient.force_authenticate(user=user)`
?

Reply:
I also tried to add 
`request = rf.get('/')`
`comment_serializer = CommentSerializer(comment_list, many=True, context={'request': request})`

as it is done in tests/categories/test_categories_filters.py

This returns:

comment_serializer.data
*** AttributeError: 'WSGIRequest' object has no attribute 'user'